tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -4226,85 +4307,97 @@ GetGenericCudnnOperationGraph(
     CudnnHandle& cudnn, std::string serialized_graph = "") {
   PreloadCudnnSubLibsHelper(kind);
 
-  // Struct to describe the ops (convolution and pointwise) in the sequence
-  // described by the graph.
-  struct SequentialOpDescriptor {
-    InputKind input_kind;
-    std::variant<cudnnConvolutionMode_t, cudnnPointwiseMode_t> mode;
-    dnn::DataType output_type;
-  };
-
-  // The format of the serialized graph describing a linear sequence of ops
-  // fused into the cuDNN convolution Custom Call is
-  // "conv[output_type]->op_name[output_type]->op_name[output_type]->..." with
-  // the convolution assumed to be first op in the graph.
-  auto deserialize_cudnn_graph =
-      [&]() -> tsl::StatusOr<std::vector<SequentialOpDescriptor>> {
-    std::vector<SequentialOpDescriptor> op_sequence = {};
+  // The format of the serialized graph describing a sequence of ops fused
+  // into the cuDNN convolution Custom Call is
+  // "UID:[output_type]conv();UID[output_type]:op_name(operand
+  // UID);UID:[output_type]op_name(operand UID);..." with the convolution
+  // assumed to be the first op in the graph. Operand UIDs identifying ops
+  // outside the serialized graph are elided.
+  auto deserialize_cudnn_graph = [&]() -> tsl::StatusOr<OpGraph> {
+    OpGraph op_graph;
     std::string::size_type pos = 0;
     while (pos < serialized_graph.size()) {
-      std::variant<cudnnConvolutionMode_t, cudnnPointwiseMode_t> mode;
+      OpMode mode;
       dnn::DataType output_type;
-      InputKind input_kind = InputKind::kNone;
       std::string::size_type m = serialized_graph.find('[', pos);
       std::string::size_type n = serialized_graph.find(']', pos);
-      std::string op_string = serialized_graph.substr(pos, m - pos);
+      int uid = std::stoi(serialized_graph.substr(pos, m - pos - 1));
       std::string data_type_string = serialized_graph.substr(m + 1, n - m - 1);
+      m = serialized_graph.find('(', pos);
+      std::string op_string = serialized_graph.substr(n + 1, m - n - 1);
+      std::optional<int> operand;
+      std::string::size_type l = serialized_graph.find(')', m + 1);
+      if (l > m + 1) {
+        operand = std::stoi(serialized_graph.substr(m + 1, l - m - 1));
+      }
+
+      if (serialized_graph.find(';', pos) != l + 1) {
```

Comment:
Instead of calling find, just check serialized_graph[l+1] != ';'.

Also DCHECK that there is an operand if and only if the op is not conv